Add Li.fi plugin

sunosuporno opened the pull request:
# Background
This PR introduces a new LiFi bridge plugin that enables cross-chain token transfers within our application. LiFi is a bridge aggregator that finds the best routes for transferring tokens across different blockchain networks, making it easier and more efficient for users to bridge their assets.

## What does this PR do?
- Implements a new `LifiService` plugin with two main tools:
  1. `lifi_get_quote`: Fetches quotes for cross-chain transfers using the LiFi API
  2. `lifi_bridge`: Executes the bridge transaction, including automatic token approvals
- Features:
  - Supports multiple EVM chains through chain mapping
  - Handles token approvals automatically before bridging
  - Includes error handling for cases when no bridge routes are available
  - Optional API key support for enhanced rate limits
  - Comprehensive logging for better debugging and user feedback

# Testing
| Method | Prompt | Screenshot | Transaction Link |
|----------|--------|-------|------------------|
| Getting Quotes | can you give quotes for swapping 1 USDC from Mode to Base? | <img width="682" alt="Screenshot 2025-03-25 at 11 40 22 PM" src="https://github.com/user-attachments/assets/37b04c89-3198-4395-88c4-868966f28f23" /> | NA |
| Bridging funds | bridge 1 USDC from Mode to Base | <img width="577" alt="Screenshot 2025-03-25 at 11 44 55 PM" src="https://github.com/user-attachments/assets/f96570bb-4a1b-4d80-a619-c9f0e35f3cc7" /> | [Transaction Link](https://explorer.mode.network/tx/0x43ef0c587709623d7ef2e31d3aa4b1a36eada259dcaa826656ce94e9f1be0e5d) |



# Docs
My changes do not require a change to the project documentation.

## Checklist
- [x] I have tested this change and added the relevant screenshots to the PR description
- [x] I updated the [README](https://github.com/goat-sdk/goat/blob/main/README.md) if necessary to include the new plugin, wallet, chain, etc.

0xaguspunk commented:
@sunosuporno could you rebase from main and remove the changes to the main README?

Also, could you add a README for the new plugin? 🙏

sunosuporno commented:
@0xaguspunk made the readme related changes! What do you mean by rebasing? I already merged all changes in main to this branch? Am I missing something?

0xaguspunk commented:
you pushed a bunch of changes to the main README in the root. I dont think these are related to the PR

sunosuporno commented:
Ohh I removed it now in the latest commit https://github.com/goat-sdk/goat/pull/450/commits/ad6228c78beb84fc94749c9e2dd35b3938e74d16. I added as the last item in the checklist of the PR wanted it 😅

0xaguspunk commented:
its failing to build, you need to delete the pnpm lock file and run pnpm install again

sunosuporno commented:
Done! @0xaguspunk

0xaguspunk commented:
@sunosuporno you pushed the changes of root project README, could you revert those? not needed for this PR :)

0xaguspunk commented:
Its still there

<img width="1396" alt="Screenshot 2025-03-25 at 21 39 58" src="https://github.com/user-attachments/assets/fb502abd-f8bf-442d-b7e3-8eb7ff90d4d1" />

sunosuporno commented:
What about now?
